src/viam/rpc/dial.py

```diff
@@ -30,7 +30,7 @@ class Credentials:
     Currently only supports robot location secret.
     """
 
-    type: Union[Literal["robot-location-secret"], Literal["robot-secret"]]
+    type: Union[Literal["robot-location-secret"], Literal["robot-secret"], Literal["api-key"]]
```

Comment:
[q] Should we add validation on this field such that users would get an error if they supplied a bad literal/typo? I'm surprised I was able to supply the `api-key` string in the original org API key [PR](https://github.com/viamrobotics/viam-python-sdk/pull/419) without hitting any errors.

Reply:
it's just a type hint right now, so no error - the idea was that users could supply other cred types as the server changes without changes in the SDK, which I think is nicer - the server can/will reject if the type is incorrect anyway. I do think we could perhaps prepopulate these types, but out of scope for now